### DISCUSSION TITLE
How to enable insecure download in Firefox while running parallel cypress tests using cypress-parallel?

### DISCUSSION THREAD

**Discussion by fardin-hossain-oyon** [cypress-io/cypress#29561] [Questions and Help]
I am running multiple cypress spec files in parallel using **'cypress-parallel'**. I added the following commands in the **'scripts'** section of the  **'package.json'** file in my project:

```
"scripts": {
    "parallelizationTest": "npx cypress run --headed --config-file ./cypress-my-config.js --browser firefox",
    "cy:parallel": "cypress-parallel -d cypress/e2e/ParallelizationTest/test/ -t 9 -s parallelizationTest"
  },
```

Then I am running my test  using `npm run cy:parallel` command.

Now, the problem is, in my tests, some files need to be downloaded from an insecure website (http://) but firefox does not allow these downloads saying insecure download is blocked. I tried changing the `dom.block_download_insecure` to false in Firefox's `about:config` section but it doesn't change anything.

The interesting thing is, when I run my tests without parallelization, the downloads are not blocked. The issue only happens in parallel tests.

Can anyone give me a fix for this issue?

**Comment by portyu9** [comment]
Hi @fardin-hossain-oyon 

Changing `about:config` manually is unreliable here because parallel Cypress runs launch their own browser processes/profiles.

Set the Firefox preference for every Cypress-launched browser instead:

```js
e2e: {
  setupNodeEvents(on, config) {
    on('before:browser:launch', (browser, launchOptions) => {
      if (browser.family === 'firefox') {
        launchOptions.preferences['dom.block_download_insecure'] = false
      }

      return launchOptions
    })
  },
}
```

That ensures each Firefox instance started by the parallel runs receives the preference.

`cypress-parallel` itself is a third-party package, but this browser configuration is handled by Cypress.

If possible, the preferable long-term fix is to serve the download over HTTPS rather than disabling Firefox's insecure-download protection.

Please mark this answer as accepted if it helped, thank you.
